core/lib/zksync_core/src/state_keeper/metrics.rs

```diff
@@ -395,3 +397,53 @@ pub(super) struct ExecutorMetrics {
 
 #[vise::register]
 pub(super) static EXECUTOR_METRICS: vise::Global<ExecutorMetrics> = vise::Global::new();
+
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "batch_tip")]
+pub(crate) struct BatchTipMetrics {
+    #[metrics(buckets = Buckets::exponential(60000.0..=80000000.0, 2.0))]
+    gas_used: Histogram<usize>,
+    #[metrics(buckets = Buckets::exponential(1.0..=60000.0, 2.0))]
+    pubdata_published: Histogram<usize>,
+    #[metrics(buckets = Buckets::exponential(1.0..=4096.0, 2.0))]
+    circuit_statistic: Histogram<usize>,
+    #[metrics(buckets = Buckets::exponential(1.0..=4096.0, 2.0))]
+    execution_metrics_size: Histogram<usize>,
+    #[metrics(buckets = Buckets::exponential(1.0..=60000.0, 2.0))]
+    block_writes_metrics_positive_size: Histogram<usize>,
+    #[metrics(buckets = Buckets::exponential(1.0..=60000.0, 2.0))]
+    block_writes_metrics_negative_size: Histogram<usize>,
+}
+
+impl BatchTipMetrics {
+    pub fn observe(&self, execution_result: &VmExecutionResultAndLogs) {
+        self.gas_used
+            .observe(execution_result.statistics.gas_used as usize);
+        self.pubdata_published
+            .observe(execution_result.statistics.pubdata_published as usize);
+        self.circuit_statistic
+            .observe(execution_result.statistics.circuit_statistic.total());
+        self.execution_metrics_size
+            .observe(execution_result.get_execution_metrics(None).size());
+    }
+
+    pub fn observe_writes_metrics(
+        &self,
+        initial_writes_metrics: &DeduplicatedWritesMetrics,
+        applied_writes_metrics: &DeduplicatedWritesMetrics,
+    ) {
+        let size_diff = applied_writes_metrics.size(ProtocolVersionId::latest()) as i128
+            - initial_writes_metrics.size(ProtocolVersionId::latest()) as i128;
```

Comment:
The latest is wrong. you have to use proper version

Reply:
And could you describe what is going on here..
i'm really confused

Reply:
Probably I should’ve documented it better. These are the difference between writes metrics before applying and deduplicating them to calculate overhead of batch tip writes metrics(see the comment of @perekopskiy above). Since in seal criteria we are using only size of them - I have shortened it to simple substraction of sizes, and to check a suggestion about negative diffs we have 2 different histograms - for negative diffs and for positive

Reply:
still using latest version is wrong, instead pass one from `updates_manager` (this is the version used for the batch)

Reply:
Updated